supabase/migrations/20240405134500_functions_to_insert_and_update_client_and_their_family.sql

```diff
@@ -0,0 +1,72 @@
+set check_function_bodies = off;
+
+CREATE OR REPLACE FUNCTION public."insertClientAndTheirFamily"(clientrecord jsonb, familymembers jsonb)
+ RETURNS uuid
+ LANGUAGE plpgsql
+AS $function$DECLARE
+    created_family_id UUID;
+    member JSONB;
+    inserted_client_id UUID;
+BEGIN  
+    INSERT INTO clients (full_name, phone_number, address_1, address_2, address_town, address_county, address_postcode, delivery_instructions, dietary_requirements, feminine_products, baby_food, pet_food, other_items, extra_information, flagged_for_attention, signposting_call_required)
+    VALUES (clientRecord->>'full_name', clientRecord->>'phone_number', clientRecord->>'address_1', clientRecord->>'address_2', clientRecord->>'address_town', clientRecord->>'address_county', clientRecord->>'address_postcode', clientRecord->>'delivery_instructions', ARRAY[clientRecord->>'dietary_requirements']::text[], ARRAY[clientRecord->>'feminine_products']::text[], CASE WHEN clientRecord->>'baby_food' = 'true' THEN TRUE
+     WHEN clientRecord->>'baby_food' = 'false' THEN FALSE

```

Comment:
does this `WHEN ___ = 'false` really work? I somehow expected this to be already boolean, so it kinda feels like we're comparing that against a string so it always chooses null? If it does work then ignore me :)

Reply:
oh dear you're right it does not work ... gonna work on fixing

Reply:
Okay nvm it does work, but i've identified some bugs im gonna fix:

-when creating or updating, the baby nappy size column needs to be treated properly (i didnt realise it existed)
-when updating only, the arrays are not processed correctly, and maybe the booleans too

Reply:
edit: think the nappy size column is treated ok

Reply:
edit: arrays not processed correctly for creating either, rip

Reply:
edit: fixed